tuf/api/metadata.py

```diff
@@ -126,25 +92,27 @@ def __update_signature(self, signatures, keyid, signature):
         if not updated:
             signatures.append(keyid_signature)
 
-    def sign(self) -> JsonDict:
+    def sign(self, key_ring: KeyRing) -> JsonDict:
+        # FIXME: Needs documentation of expected behavior
         signed_bytes = self.signed_bytes
```

Comment:
unused variable?

Reply:
Oh, @trishankatdatadog passed `signed_bytes` to `key.sign`. I didn't understand why (same goes for `signatures = self.__signatures`) and started changing it, but then though it might be better to ask him why he did it that way. :) @trishankatdatadog, could you help me understand the `sign` and `verify` functions?

Reply:
You mean `Metadata.(sign|verify)`? To call `Key.(sign|verify)`, really, it should not be concerned with how keys actually do the signing. I have no strong opinions on where `signed` and `signatures` come from, please feel free to adjust according to your new implementation.